src/main.py

```diff
@@ -1,13 +1,98 @@
-from fastapi import FastAPI
+"""
+Main application module for the Spark Manager API.
+"""
 
-app = FastAPI(title="CDM Spark Cluster Manager API")
+import logging
 
+from fastapi import FastAPI, HTTPException, Request, Response
+from fastapi.exceptions import RequestValidationError
+from fastapi.middleware.gzip import GZipMiddleware
+from fastapi.security.utils import get_authorization_scheme_param
+from starlette.middleware.base import BaseHTTPMiddleware
 
-@app.get("/health")
-async def health_check():
-    """Health check endpoint."""
-    return {"status": "healthy"}
+from src.routes import health
+from src.service import app_state
+from src.service.config import configure_logging, get_settings
+from src.service.exception_handlers import universal_error_handler
+from src.service.exceptions import InvalidAuthHeaderError, SparkManagerError
+from src.service.models import ErrorResponse
 
-if __name__ == "__main__":
-    import uvicorn
-    uvicorn.run(app, host="0.0.0.0", port=8000) 
\ No newline at end of file
+# Configure logging
+configure_logging()
+logger = logging.getLogger(__name__)
+
+# Middleware constants
+_SCHEME = "Bearer"
+
+
+class AuthMiddleware(BaseHTTPMiddleware):
+    """Middleware to authenticate users and set them in the request state."""
+
+    async def dispatch(self, request: Request, call_next) -> Response:
+        request_user = None
+        auth_header = request.headers.get("Authorization")
+
+        if auth_header:
+            scheme, credentials = get_authorization_scheme_param(auth_header)
+            if not (scheme and credentials):
+                raise InvalidAuthHeaderError(
+                    f"Authorization header requires {_SCHEME} scheme followed by token"
+                )
+            if scheme.lower() != _SCHEME.lower():
+                # don't put the received scheme in the error message, might be a token
+                raise InvalidAuthHeaderError(
+                    f"Authorization header requires {_SCHEME} scheme"
+                )
+
+            app_state_obj = app_state.get_app_state(request)
+            request_user = await app_state_obj.auth.get_user(credentials)
+
+        app_state.set_request_user(request, request_user)
+
+        return await call_next(request)
+
+
+def create_application() -> FastAPI:
+    """Create and configure the FastAPI application."""
+    settings = get_settings()
+
+    app = FastAPI(
+        title=settings.app_name,
+        description=settings.app_description,
+        version=settings.api_version,
+        responses={
+            "4XX": {"model": ErrorResponse},
+            "5XX": {"model": ErrorResponse},
+        },
+    )
+
+    # Add exception handlers
+    app.add_exception_handler(RequestValidationError, universal_error_handler)
+    app.add_exception_handler(HTTPException, universal_error_handler)
+    app.add_exception_handler(SparkManagerError, universal_error_handler)
+    app.add_exception_handler(Exception, universal_error_handler)
```

Comment:
All those inherit from Exception right? Since they're all going to the same error handler I think you only need the last line

Reply:
👍